lib/common/xml.c

```diff
@@ -604,17 +604,39 @@ pcmk__xe_remove_matching_attrs(xmlNode *element,
 
 /*!
  * \internal
- * \brief Create a new XML element under a given parent
+ * \brief Set a given string as an XML node's content
+ *
+ * \param[in,out] node     Node whose content to set
+ * \param[in]     content  String to set as the content
+ *
+ * \note \c xmlNodeSetContent() does not escape special characters.
+ */
+void
+pcmk__xe_set_content(xmlNode *node, const char *content)
+{
+    if (node != NULL) {
+        char *escaped = pcmk__xml_escape(content, false);
+
+        xmlNodeSetContent(node, (pcmkXmlStr) escaped);
+        free(escaped);
+    }
+}
+
+/*!
+ * \internal
+ * \brief Create a new XML element under a given parent with text content
  *
- * \param[in,out] parent  XML element that will be the new element's parent
- *                        (\c NULL to create a new XML document with the new
- *                        node as root)
- * \param[in]     name    Name of new element
+ * \param[in,out] parent   XML element that will be the new element's parent
+ *                         (\c NULL to create a new XML document with the new
+ *                         node as root)
+ * \param[in]     name     Name of new element
+ * \param[in]     content  Text to set as the new element's content (can be
+ *                         \c NULL)
  *
  * \return Newly created XML element, or \c NULL on memory allocation failure
  */
 xmlNode *
-pcmk__xe_create(xmlNode *parent, const char *name)
+pcmk__xe_create_full(xmlNode *parent, const char *name, const char *content)
```

Comment:
We can also think about adding a variadic argument to the end, to allow setting attributes at creation time like we do with `pcmk__output_create_xml_node()`. I didn't bother right now